extensions/vscode/src/multiStepInputs/newConnectCredential.ts

```diff
@@ -490,15 +506,17 @@ export async function newConnectCredential(
     input: MultiStepInput,
     state: MultiStepState,
   ) {
-    // url should always be defined by the time we get to this step
-    // but we have to type guard it for the API
+    // url and apiKey should always be defined by the time we get to this step
+    // but we have to type guard them for the API
     const serverUrl = typeof state.data.url === "string" ? state.data.url : "";
+    const apiKey =
+      typeof state.data.apiKey === "string" ? state.data.apiKey : "";
     let connections: SnowflakeConnection[] = [];
     let connectionQuickPicks: QuickPickItemWithIndex[] = [];
 
     try {
-      await showProgress("Reading Snowflake connections", viewId, async () => {
-        const resp = await fetchSnowflakeConnections(serverUrl);
+      await showProgress("Testing Snowflake connections", viewId, async () => {
+        const resp = await fetchSnowflakeConnections(serverUrl, apiKey);
```

Comment:
Looks like we always send up an `apiKey` so no worries removing the optional in the other spots noted.